### PR TITLE
etcdserverpb: document how to prefix, range query

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5401.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @heyitsanthony @xiang90 Is this the right place to document how etcd server side processes range requests? I think client side, we have enough documentation using godoc.

And what's confusing me, we have different notations, and I am not familiar with some of them. We have `noPrefixEnd = []byte{0}` later to be replaced by `[]byte{}` in server side, string literal `"\x00"` in Go code to be used with range_end `"\x00"` to range all keys, and we have `'\0'` in protocol buffer documentation for this `\x00`. So I am not sure if I am using these notations correctly.

Can you take a look at this PR?

I will regenerate with protodoc after this gets reviewed.

Thanks.

File: etcdserver/etcdserverpb/rpc.proto [hunk 1]
```diff
@@ -162,10 +162,14 @@ message RangeRequest {
 	VALUE = 4;
   }
 
-  // key is the first key for the range. If range_end is not given, the request only looks up key.
+  // key is the first key for the range.
   bytes key = 1;
   // range_end is the upper bound on the requested range [key, range_end).
-  // If range_end is '\0', the range is all keys >= key.
```
Comment (xiang90) [inline etcdserver/etcdserverpb/rpc.proto]: the old line 168 is correct, we should kill the new line168 and line 169.

for line 170,

If the `range_end` is one bit larger than the given `key`, then the range requests get the all keys with the prefix `key`

File: etcdserver/etcdserverpb/rpc.proto [hunk 2]
```diff
@@ -162,10 +162,14 @@ message RangeRequest {
...
+  // If range_end is not given or []bytes{0}, the range is defined to contain only the key.
+  // If range_end is '\x00', the range is all keys greater than or equal to the key.
+  // If key is given and range_end is same as key except the last byte that is one-byte greater than key's last byte,
+  // the range is all keys with key as their prefixes.
+  // If both key and range_end are '\x00', the range is all keys.
```
Comment (xiang90) [inline etcdserver/etcdserverpb/rpc.proto]: If both key and range_end are `\0`, then range requests returns all keys.

**Comment by heyitsanthony** [comment]
lgtm once xiang's comments are addressed

File: etcdserver/etcdserverpb/rpc.proto [hunk 3]
```diff
@@ -166,6 +166,9 @@ message RangeRequest {
   bytes key = 1;
   // range_end is the upper bound on the requested range [key, range_end).
   // If range_end is '\0', the range is all keys >= key.
+  // If the range_end is one bit larger than the given key,
+  // then the range requests get the all keys with the prefix key.
```
Comment (xiang90) [inline etcdserver/etcdserverpb/rpc.proto]: with the prefix (the given key)

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
Just fixed. Merging in greens. Thanks!

**Pull request merged**